notifications_utils/recipient_validation/phone_number.py

```diff
@@ -225,6 +226,10 @@ def validate_phone_number(self, phone_number: str) -> phonenumbers.PhoneNumber:
 
         return number
 
+    @staticmethod
+    def _is_premium_number(phone_number) -> bool:
+        return phonenumbers.number_type(phone_number) == phonenumbers.PhoneNumberType.PREMIUM_RATE
```

Comment:
as mentioned before, probably good to flip this over to be `def _is_landline_or_mobile(phone_number: PhoneNumber) -> bool:` (i'll leave checking what the PhoneNumberType enum actually contains to you, and to check if 01x, 02x, 03x and 07x numbers have different types in any way)

Reply:
I have moved this into an allow and a deny list:

```
ALLOW_LIST = [
    phonenumbers.PhoneNumberType.FIXED_LINE,
    phonenumbers.PhoneNumberType.MOBILE,
    phonenumbers.PhoneNumberType.FIXED_LINE_OR_MOBILE,
    phonenumbers.PhoneNumberType.UAN,
    phonenumbers.PhoneNumberType.TOLL_FREE,
    phonenumbers.PhoneNumberType.PERSONAL_NUMBER,
]

DENY_LIST = [
    phonenumbers.PhoneNumberType.PREMIUM_RATE,
]
```

This is currently comprehensive for the different number types we use in our tests (we need all of these in the allow list). It _may_ be that we don't want to be this permissive (benefit that this new validation code gives us better visibility about exactly what types of numbers we're allowing). I don't know if that's a question for this ticket or not.